windbow27/Kornblume
# Feat: i18n POC

## Description

Here is a POC of feature for i18n.

## Changes
- Install [vue-i18n](https://vue-i18n.intlify.dev/guide/introduction.html)
- Add language JSON files
- Add a language switch to the Navbar, controlled by a feature toggle

## How to Test
Install dependencies by `npm i` before run the dev server

This feature is controlled by a feature toggle
so if you want to enable it, you need to key-in a command through the console:
```
localStorage.setItem('i18n-testing', 1)
```

After refreshing, you should be able to see the language switch (for testing) on the Navbar

![image](https://github.com/windbow27/Kornblume/assets/26058569/02b92ad8-f86d-41a0-bf80-7df5b735e280)

You can also play around on the [playground](https://frantw.github.io/Kornblume/) site

## How to Help Set Up i18n Keys

Right now, the hard-coded words on our site don't have their i18n keys yet
So, whenever you've got some free time, feel free to jump in and help out with that

Installing the [i18n Ally](https://marketplace.visualstudio.com/items?itemName=lokalise.i18n-ally) can make this whole process a lot easier

![i18n_ally](https://github.com/windbow27/Kornblume/assets/26058569/cdedaf91-f069-4032-802c-7135373cd5ed)

**Don't forgot to set `en-US` as source language!**
